src/chains.rs

```diff
@@ -284,7 +284,7 @@ impl Rewrite for ChainItem {
             ChainItemKind::StructField(ident) => format!(".{}", rewrite_ident(context, ident)),
             ChainItemKind::TupleField(ident, nested) => format!(
                 "{}.{}",
-                if nested && context.config.version() == Version::One {
+                if nested && context.config.style_edition() <= StyleEdition::Edition2021 {
```

Comment:
General comment to review these carefully, tedious as it may be. Version One is the analog of Style Edition 2015-2021 and reflected as such with the `<=` operator, Version Two `>=` on Style 2024.

We don't have an exhaustive suite of tests for each and every possible syntactical scenarios that may trigger these code paths, so important to double and triple check for human error (I did myself several times, second pair of eyes appreciated).